whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/inetnum/IntersectionValidator.java

```diff
@@ -39,11 +40,10 @@ public IntersectionValidator(final Ipv4Tree ipv4Tree, final Ipv6Tree ipv6Tree) {
     @Override
     public List<Message> performValidation(final PreparedUpdate update, final UpdateContext updateContext) {
         final IpInterval ipInterval = IpInterval.parse(update.getReferenceObject().getKey());
-        if (ipInterval instanceof Ipv4Resource) {
-            return validateIntersections(ipInterval, ipv4Tree, Lists.newArrayList());
-        } else {
-            return validateIntersections(ipInterval, ipv6Tree, Lists.newArrayList());
-        }
+        return switch (ipInterval) {
+            case Ipv4Resource ipv4Resource -> validateIntersections(ipInterval, ipv4Tree, Lists.newArrayList());
+            case Ipv6Resource ipv6Resource -> validateIntersections(ipInterval, ipv6Tree, Lists.newArrayList());
+        };
```

Comment:
Do you need null case here as well?

Reply:
No because the original code was an if-else with no null check